logr.go

```diff
@@ -395,51 +395,50 @@ func (l Logger) WithContext(ctx context.Context) Logger {
 	return l
 }
 
-// WithContextValues extends the list of context keys and the name for them. When
-// given a context through WithContext later, the Logger will extract the
-// values for these keys and log them as additional key/value pairs
+// WithContextValues extends the list of context handlers. When
+// given a context through WithContext later, the Logger will invoke
+// each handler and log the key/value pairs that they return as
 // as if they had been passed to WithValues.
-func (l Logger) WithContextValues(keys ...ContextKey) Logger {
+func (l Logger) WithContextValues(handlers ...ContextHandler) Logger {
 	if l.sink == discardLogSinkInstance {
 		return l
 	}
 
-	if l.contextKeys == nil {
+	if l.contextHandlers == nil {
 		// Copy the parameters to avoid surprises when the caller changes the content
 		// of the parameter slice later.
-		cp := make([]ContextKey, len(keys))
-		copy(cp, keys)
-		l.contextKeys = &cp
+		cp := make([]ContextHandler, len(handlers))
+		copy(cp, handlers)
+		l.contextHandlers = &cp
 	} else {
 		// We must create a new slice because the existing one is shared between
 		// Logger instances.
-		cp := make([]ContextKey, len(*l.contextKeys)+len(keys))
-		copy(cp, *l.contextKeys)
-		copy(cp[len(*l.contextKeys):], keys)
-		l.contextKeys = &cp
+		cp := make([]ContextHandler, len(*l.contextHandlers)+len(handlers))
```

Comment:
you may have been correct that append was easier to read :)

Reply:
I'll go back ;-}